src/components/Checkbox/Checkbox.js

```diff
@@ -161,11 +162,27 @@ const CheckboxComponent = (
     disabled,
     validationHint,
     className,
+    tracking,
     ...props
   },
   ref
 ) => {
   const id = customId || uniqueId('checkbox_');
+  const { label, component = 'checkbox', customParameters } = tracking || {};
+  const dispatch = useClickTrigger();
+  const handleChange =
+    props.onChange && label
+      ? e => {
+          dispatch({
+            label,
+            component,
+            customParameters
+          });
+
+          props.onChange(e);
+        }
+      : props.onChange;
+
```

Comment:
Seeing this code in every component, I'm starting to wonder whether we should add a helper function for this:

```ts
function createChangeHandler(onChange, dispatch, tracking = {}, defaultComponent) {
  const { label, component = defaultComponent, customParameters } = tracking;

  if (!label) {
    return onChange;
  }

  return (event) => {
    dispatch({ label, component, customParameters });

    if (onChange) {
      onChange(event);
    }
  };
}

// Usage:
const handleChange = createChangeHandler(onChange, dispatch, tracking, 'checkbox');
```

It would clean up the component code, you'd only need to test the `onChange` handler creation in one place and it's easier to evolve the code in the future. The handler would live in the `src/util` folder.

What do you think?

Reply:
Absolutely make sense. I was thinking about this too. Just wanna finish reviewing components first and collect enough use cases to create proper abstraction.

Just one small thing. Do you think we should leave `dispatch` overwritten here? I'm thinking that since this function is a helper for default click/change callback, maybe make sense to just use `useClickTrigger ` here?

Reply:
Yeah, that could be even better. In that case, the abstraction would need to be a custom hook.

I wasn't sure if there could be different types of `dispatch` (click vs view vs change).

Reply:
For clicking and changing, it would be the same, but for viewing, it would be a different hook I suppose. So I think there might be 2 different helpers: `createChangingHandler` and `createViewingHandler`.